app/decorators/account_decorator.rb

```diff
@@ -3,9 +3,30 @@
 class AccountDecorator < ApplicationDecorator
   delegate :display_name, :email, to: :admin_user, prefix: true
 
+  self.include_root_in_json = false
+
+  def new_application_data
+    {
+      id: id.to_s,
+      name: name,
+      description: "Admin: #{admin_user_display_name}",
+      createdAt: created_at.to_s(:long),
+      contractedProducts: contracts,
+      createApplicationPath: h.admin_buyers_account_applications_path(object),
+      multipleAppsAllowed: multiple_applications_allowed?
+    }
+  end
+
   private
 
   def admin_user
     @admin_user ||= (super || User.new).decorate
   end
+
+  def contracts
+    bought_service_contracts.map do |contract|
+      hash = contract.service.as_json(only: %i[id name], root: false)
+      hash['withPlan'] = contract.plan.as_json(only: %i[id name], root: false)
```

Comment:
Are you sure you want the single value `hash['withPlan'` and not the entire `hash` in the Array?

Reply:
It's actually mapping the hash `id` and `name` 🤔 withPlan is lost and still the logic doesn't break. Maybe I added some redundancy? Could you check the following logic makes sense?
https://github.com/3scale/porta/blob/e24aca233b76b5f727b065ac4095329247c2f27e/app/javascript/src/Logic/BuyerLogic.js#L12-L15